spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/DataTables.groovy

```diff
@@ -446,100 +252,235 @@ local ; 1
+    def results = runner.runFeatureBody '''
       expect:
-      false
+      true
+      a == b
+      c == "x$i"
 
       where:
-      a | b
-      1 | b + 1
+      x    | _
+      'x1' | _
+      'x2' | _
+      'x3' | _
+      'x4' | _
+      'x5' | _
+      'x6' | _
+
+      i << (1..6)
+
+      a | _
+      1 | _
+      2 | _
+      combined:
+      b | _
+      a | _
+      combined:
+      c | _
+      x | _
+      x | _
+      x | _
     '''
 
     then:
-    thrown Exception
+    results.testsSucceededCount == 1 + 6
   }
 
-  def 'data tables can be referenced from following variables'() {
-    expect:
-    c == 3
+  def 'data tables with different widths can be combined'() {
+    when:
+    def results = runner.runSpecBody '''
+      def 'a feature (#a #b #c #d #e)'() {
+        expect:
+        true
 
-    where:
-    a | b
-    1 | 2
-    1 | a + 1
+        where:
+        a | b
+        1 | 'b'
+        2 | 'b'
+        combined:
+        c | d   | e
+        3 | 'd' | 'e'
+      }
+    '''
 
-    c = b + 1
+    then:
+    results.testsStartedCount == 1 + 2
+    results.testEvents().started().list().testDescriptor.displayName == [
+      'a feature (#a #b #c #d #e)',
+      'a feature (1 b 3 d e)',
+      'a feature (2 b 3 d e)'
+    ]
   }
 
-  def "rows must have same number of elements as header"() {
+  def 'data tables starting with table separator can be combined'() {
     when:
-    compiler.compileFeatureBody """
-expect:
-true
+    def results = runner.runSpecBody '''
+      def 'a feature (#a #b #c #d #e)'() {
+        expect:
+        true
 
-where:
-a | b | c
-1 | 2 | 3
-4 | 5
-    """
+        where:
+        a | b
+        1 | 'b'
+        2 | 'b'
+        combined:
+        _____________
+        c | d   | e
+        3 | 'd' | 'e'
+      }
+    '''
 
     then:
-    thrown(InvalidSpecCompileException)
+    results.testsStartedCount == 1 + 2
+    results.testEvents().started().list().testDescriptor.displayName == [
+      'a feature (#a #b #c #d #e)',
+      'a feature (1 b 3 d e)',
+      'a feature (2 b 3 d e)'
+    ]
   }
 
-  def "rows must have same number of elements as header with semicolon"() {
+  def 'data tables and data pipes can be combined'() {
     when:
-    compiler.compileFeatureBody """
-expect:
-true
+    def results = runner.runSpecBody '''
+      def 'a feature (#a #b #c)'() {
+        expect:
+        true
 
-where:
-a ; b ; c
-1 ; 2 ; 3
-4 ; 5
-    """
+        where:
+        a | b
+        1 | 'b'
+        2 | 'b'
+        combined:
+        c << [3]
+      }
+    '''
 
     then:
-    thrown(InvalidSpecCompileException)
+    results.testsStartedCount == 1 + 2
+    results.testEvents().started().list().testDescriptor.displayName == [
+      'a feature (#a #b #c)',
+      'a feature (1 b 3)',
+      'a feature (2 b 3)'
+    ]
   }
 
-  def "cells can be separated with single or double pipe operator"() {
-    expect:
-    a + b == c
+  def 'data pipes and data tables can be combined'() {
```

Comment:
What about data processors at the end?

Reply:
Could you elaborate on what you mean?
I'm pretty unsure what you try to ask.
Data processors are an implementation detail.
Or do you mean that you miss a derived data variable that accesses multiplied variables?
If so, I added one now to the `DataSpec`, so it also is shown in the docs.